examples/http_async_server.hpp

```diff
@@ -83,59 +212,81 @@ class http_async_server
         peer& operator=(peer const&) = delete;
 
         explicit
-        peer(socket_type&& sock, std::string const& root)
-            : stream_(std::move(sock))
-            , strand_(stream_.get_io_service())
-            , root_(root)
+        peer(socket_type&& sock, http_async_server& server)
```

Comment:
I don't believe this multi-argument constructor needs to be marked `explicit`, but there's no harm to it.